src/DittoMachine.sol

```diff
@@ -342,24 +343,17 @@ contract DittoMachine is ERC721, ERC721TokenReceiver {
     }
 
     ////////////////////////////////////////////////
-    ////////////// EXTERNAL FUNCTIONS //////////////
+    ////////////// RECEIVER FUNCTIONS //////////////
     ////////////////////////////////////////////////
 
-    /**
-     * @dev will allow NFT sellers to sell by safeTransferFrom-ing directly to this contract.
-     * @param data will contain ERC20 address that the seller wishes to sell for
-     * allows specifying selling for the floor price
-     * @return returns received selector
-     */
-    function onERC721Received(
-        address,
+    function onTokenReceived(
         address from,
+        address ERC721Contract,
         uint256 id,
-        bytes calldata data
-    ) external returns (bytes4) {
-        address ERC721Contract = msg.sender;
-        (address ERC20Contract, bool floor) = abi.decode(data, (address, bool));
-
+        address ERC20Contract,
+        bool floor,
+        bool ERC1155Contract
```

Comment:
would recommend changing this to `isERC1155Contract`. From our naming convention, it's easy to assume that it's an `address`.

Reply:
yeah, good call, will fix in next commit.